### DISCUSSION TITLE
Conflict Between Python httpx Module and Project Discovery "httpx" Tool

### DISCUSSION THREAD

**Discussion by GDATTACKER-RESEARCHER** [encode/httpx#3459] [General]
Dear team,

I am currently encountering a conflict between the Python module named httpx and the "httpx" tool developed by Project Discovery. Both share the same name, which is causing issues when using them together, especially when running tools that depend on either the tool or the module.

When the Python httpx module is installed, it interferes with the execution of the "httpx" tool by Project Discovery, leading to errors or unexpected behavior. This issue becomes more apparent when tools, such as bbot, rely on one of them but don’t specify which one to use.

Here are the steps to reproduce the issue:

Install the Python httpx module using pip install httpx.
Attempt to run the "httpx" tool from Project Discovery via the command line.
You will notice that the Python module interferes with the tool, causing either errors or unexpected behavior.
The expected behavior is to resolve the conflict between the two so that they can be used together without issues. It would be helpful if there were a way to differentiate between the "httpx" tool and the Python module or a solution to prevent interference between them.

For reference, here are the GitHub links for both:

https://github.com/encode/httpx
https://github.com/projectdiscovery/httpx
I look forward to hearing any suggestions or solutions you may have.

**Comment by zanieb** [comment]
> You will notice that the Python module interferes with the tool, causing either errors or unexpected behavior.

What sort of errors?

**Reply by GDATTACKER-RESEARCHER** [reply]
either you or projectdiscovery need to change tool name i guess because both tools are much known in community
